### PR TITLE
feat(compiler-cli): make enableIvy ngtsc/true equivalent

### PR DESCRIPTION
**Pull request by filipesilva**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?
Currently setting `enableIvy` to true runs a hybrid mode of `ngc` and `ngtsc`. This is counterintuitive given the name of the flag itself.

## What is the new behavior?
This PR makes the `true` value equivalent to the previous `ngtsc`, and `ngtsc` becomes an alias for `true`. Effectively this removes the hybrid mode as well since there's no other way to enable it.

## Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

## Other information
I'm not sure if the `true` option is currently used anywhere. 

@alxhub can you advise on:
- ways to test this change
- if there should be a migration path documented
- if a new value should be added to `enableIvy` to keep the existing hybrid behaviour (the current `true` value)

### CODE REVIEW

**Comment by JoostK** [comment]
Related: #27306

**Comment by alxhub** [comment]
> ways to test this change

Switch `ng_module.bzl` to use the `true` value instead of `"ngtsc"`.

* if there should be a migration path documented

Nah, nothing depends on the old behavior afaik.

* if a new value should be added to `enableIvy` to keep the existing hybrid behaviour (the current `true` value)

Nope, we'll remove it soon.

File: packages/compiler-cli/src/perform_compile.ts [hunk 1]
```diff
@@ -120,6 +120,10 @@ export function calcProjectFileAndBasePath(project: string):
 
 export function createNgCompilerOptions(
     basePath: string, config: any, tsOptions: ts.CompilerOptions): api.CompilerOptions {
+  // enableIvy `true` is an alias for `ngtsc`.
+  if (config.angularCompilerOptions && config.angularCompilerOptions.enableIvy === true) {
```
Comment (alxhub) [inline packages/compiler-cli/src/perform_compile.ts]: I'd rather go the other way. `ngtsc` should be an alias for `true`.

**Review by alxhub — CHANGES_REQUESTED** [review]

File: packages/compiler-cli/src/perform_compile.ts (hunk 1 above)
Comment (filipesilva) [reply packages/compiler-cli/src/perform_compile.ts]: Done.

**Review by alxhub — APPROVED** [review]

**Review by alexeagle — APPROVED** [review]

**Comment by alexeagle** [comment]
[presubmit](http://test/OCL:234478780:BASE:234478815:1550503599822:26853ef1)

**Pull request closed without merging**